src/index.js

```diff
@@ -2,25 +2,30 @@
 
 const bs58 = require('bs58')
 
-module.exports = PeerBook
-
-function PeerBook () {
-  if (!(this instanceof PeerBook)) {
-    return new PeerBook()
+class PeerBook {
+  constructor () {
+    this._peers = {}
   }
 
-  const peers = {}
-
-  this.put = (peerInfo, replace) => {
-    if (peers[peerInfo.id.toB58String()] && !replace) {
-      // peerInfo.replace merges by default
-      peers[peerInfo.id.toB58String()].multiaddr.replace([], peerInfo.multiaddrs)
+  /**
+   * Stores a peerInfo, if already exist, only adds the multiaddrs
+   *
+   * @param {PeerInfo} peerInfo
+   * @param {replace} boolean
+   * @returns {null}
+   */
+  put (peerInfo, replace) {
+    if (this._peers[peerInfo.id.toB58String()] && !replace) {
+      // peerInfo.replace merges by default if none to replace are passed
+      this._peers[peerInfo.id.toB58String()]
+        .multiaddrs.replace([], peerInfo.multiaddrs.toArray())
```

Comment:
this does not do what we want, as I said before what I need is to merge both sets of multiaddrs, which this does not do in my tests